app/controllers/posts_controller.rb

```diff
@@ -1,4 +1,50 @@
 class PostsController < ApplicationController
+  before_action :authenticate_user!
+
   def index
+    @posts = Post.all.includes(:user).order(created_at: :desc)
+  end
+
+  def new
+    @post = Post.new
+  end
+
+  def create
+    @post = current_user.posts.build(post_params)
+    if @post.save
+      redirect_to @post, notice: '投稿が作成されたよ！'
+    else
+      puts @post.errors.full_messages
+      render :new
+    end
+  end
+
+  def show
+    @post = Post.find(params[:id])
+  end
+
+  def edit
+    @post = Post.find(params[:id])
+  end
+
+  def update
+    @post = Post.find(params[:id])
+    if @post.update(post_params)
+      redirect_to @post, notice: '更新完了だよ！'
+    else
+      render :edit
+    end
+  end
+
+  def destroy
+    @post = Post.find(params[:id])
+    @post.destroy
```

Comment:
削除できなかった時に例外を発生させるのも検討して良いかもです
このままだと削除できなかった時も素通りしてしまいます
なので、ユーザーからすると、削除したはずなのに削除されてないという事態が最悪発生してしまいます


```suggestion
    @post.destroy!
```